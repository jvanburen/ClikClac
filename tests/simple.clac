: dup 1 pick ;
: end print quit nop ;
: compare dup 100 < ;
: is1 1 print ;
: is2 2 print ;
: checkif2 dup 2 - if nop else is2 ;
: checkif1or2 dup 1 - if checkif2 else is1 ;
: nop ;
: conditional compare if dup else end ;
: inc 1 + ;
: next dup dup inc if next else conditional ;
: main 0 input checkif1or2 next ;