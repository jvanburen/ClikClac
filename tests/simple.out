MACRO_dup:
  push 1 
   pick
#____________________
MACRO_inc:
  push 1 
   add 
#____________________
MACRO_nop:
  
#____________________
MACRO_compare:
  call MACRO_dup
  push 100 
   cmp 
#____________________
MACRO_end:
   output 
  abort 
  call MACRO_nop
#____________________
MACRO_conditional:
  call MACRO_compare
  jmp LABEL1
  call MACRO_dup
  jmp LABEL2
  @label LABEL1:
  call MACRO_end
  @label LABEL2:
#____________________
MACRO_next:
  call MACRO_dup
  call MACRO_dup
  call MACRO_inc
  jmp LABEL3
  call MACRO_next
  jmp LABEL4
  @label LABEL3:
  call MACRO_conditional
  @label LABEL4:
#____________________
MAIN:
  push 0 
  call MACRO_next
#____________________
